### PR TITLE
Add bag.to_avro

### PR DESCRIPTION
**Pull request by martindurant**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

File: dask/bag/avro.py [hunk 1]
```diff
@@ -147,3 +148,69 @@ def read_file(fo):
...
+    """
+    # TODO infer schema from first partition of data
+    from .core import merge
+    from dask.utils import import_required
+    from dask import delayed, compute
+    from dask.bytes.core import open_files, logical_size
+    from dask.bag import from_delayed
+    import_required('fastavro',
+                    "fastavro is a required dependency for using "
+                    "bag.read_avro().")
```
Comment (mrocklin) [inline dask/bag/avro.py]: read_avro -> to_avro

File: dask/bag/avro.py [hunk 2]
```diff
@@ -155,6 +155,25 @@ def to_avro(b, filename, schema, name_function=None, storage_options=None,
...
+         'type': 'record',
+         'fields': [
+            {'name': 'a', 'type': 'int'},
+         ]}
+
+    where the "name" field is required, but "namespace" and "doc" are optional
+    descriptors; "type" must always be "record". The list of fields should
+    have an entry for every key of the input records, and the types are
+    like the primitive, complex or logical types of the Avro spec
+    ( https://avro.apache.org/docs/1.8.2/spec.html ).
```
Comment (mrocklin) [inline dask/bag/avro.py]: How hard would it be to create an examples section like the following?

```
Examples
--------
>>> b = db.from_sequence([{'name': 'Alice', 'value': 100}, {'name': 'Bob', 'value': 200}])
>>> schema = ...  # crazy long blob
>>> b.to_avro('my-data.*.avro', schema)  # doctest: +SKIP
```

**Comment by mrocklin** [comment]
I think that we should add docs to `docs/source/bag-api.rst` under both "Create Bags" and "Turn Bags into other things" and also maybe a brief section in `bag-creation.rst` after `db.read_text`?

**Comment by martindurant** [comment]
OK, will do

**Comment by mrocklin** [comment]
Looks like there is a doctest failure:

```
_______________________ [doctest] dask.bag.avro.to_avro ________________________
198         Included in the file header
199     compute: bool
200         If True, files are written immediately, and function blocks. If False,
201         returns delayed objects, which can be computed by the user where
202         convenient.
203     kwargs: passed to compute(), if compute=True
204 
205     Examples
206     --------
207     >>> b = db.from_sequence([{'name': 'Alice', 'value': 100},
UNEXPECTED EXCEPTION: NameError("name 'db' is not defined",)
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/doctest.py", line 1330, in __run
    compileflags, 1), test.globs)
  File "<doctest dask.bag.avro.to_avro[0]>", line 1, in <module>
NameError: name 'db' is not defined
```

**Comment by mrocklin** [comment]
Looks good to me.  Merging.  Thanks @martindurant

**Pull request merged**